test/server/graphql/v2/mutation/AccountMutations.test.ts

```diff
@@ -446,4 +449,170 @@ describe('server/graphql/v2/mutation/AccountMutations', () => {
+
+    after(() => sandbox.restore());
+
+    afterEach(() => {
+      sendEmailSpy.resetHistory();
+    });
+
+    it('sends the message by email', async () => {
+      const result = await graphqlQueryV2(
+        sendMessageMutation,
+        {
+          account: { id: idEncode(collectiveWithContact.id, 'account') },
+          message,
+          subject: 'Testing',
+        },
+        randomUser,
+      );
+
+      expect(result.errors).to.not.exist;
+      expect(result.data.sendMessage.success).to.equal(true);
+
+      await waitForCondition(() => sendEmailSpy.callCount === 1);
+      expect(sendEmailSpy.callCount).to.equal(1);
+      expect(sendEmailSpy.args[0][1]).to.equal(
+        `New message from ${randomUser.collective.name} on Open Collective: Testing`,
+      );
+      expect(sendEmailSpy.args[0][2]).to.include(message);
+    });
+
+    it('cannot inject code in the email (XSS)', async () => {
+      const code = '<script>console.log("XSS")</script>';
+      const xssUser = await fakeUser({ name: 'XSS User' }, { name: 'XSS Collective', slug: 'xss-collective' });
+      const result = await graphqlQueryV2(
+        sendMessageMutation,
+        {
+          account: { id: idEncode(collectiveWithContact.id, 'account') },
+          message: message + code,
+          subject: 'Testing',
+        },
+        xssUser,
+      );
+
+      expect(result.errors).to.not.exist;
+      expect(result.data.sendMessage.success).to.equal(true);
+
+      await waitForCondition(() => sendEmailSpy.callCount === 1);
+
+      const expectedMessage =
+        'Hi Test Collective,\n' +
+        '\n' +
+        'XSS Collective [http://localhost:3000/xss-collective] just sent you a message on\n' +
+        'Open Collective. Simply reply to this email to reply to the sender.\n' +
+        '\n' +
+        '\n' +
+        '\n' +
+        '\n' +
+        'Subject\n' +
+        '\n' +
+        '> Testing\n' +
+        '\n' +
+        '\n' +
+        'Message\n' +
+        '\n' +
+        '> Hello collective, I am reaching out to you for testing purposes.\n' +
+        '\n' +
+        'If this message is spam, please forward it to support@opencollective.com\n' +
+        '[support@opencollective.com].\n' +
+        '\n' +
+        '[http://localhost:3000/static/images/email/logo-email-footer@2x.png]\n' +
+        '[http://localhost:3000]\n' +
+        '\n' +
+        'We can do great things together\n' +
+        '\n' +
+        'You can also follow us on Twitter [https://twitter.com/OpenCollect] or come chat\n' +
+        'with us on our public Slack channel [https://slack.opencollective.com].\n' +
+        '\n' +
+        'Made with ❤️ from all over the world\n' +
+        '[https://docs.opencollective.com/help/about/team]';
+
+      expect(sendEmailSpy.callCount).to.equal(1);
+      expect(sendEmailSpy.args[0][1]).to.equal(`New message from XSS Collective on Open Collective: Testing`);
+      expect(sendEmailSpy.args[0][3].text).to.equal(expectedMessage);
```

Comment:
:no_entry_sign: Looking at `text` is not the proper way to test this: this property is generated from https://github.com/opencollective/opencollective-api/blob/a2018297a70ef372addc44290b5662562409f7af/scripts/compile-email.js#L581 by stripping all HTML content. In other works, it will never contain HTML content.

We should rather look at `html`:
```suggestion
      expect(sendEmailSpy.args[0][3].html).to.equal(expectedMessage);
```

Reply:
Updated to check the HTML content.